staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/admission.go

```diff
@@ -196,14 +204,52 @@ func (a *GenericAdmissionWebhook) Admit(attr admission.Attributes) error {
 	hooks := hookConfig.ExternalAdmissionHooks
 	ctx := context.TODO()
 
-	errCh := make(chan error, len(hooks))
+	hookIndexCh := make(chan int, len(hooks))
 	wg := sync.WaitGroup{}
 	wg.Add(len(hooks))
 	for i := range hooks {
+		go func(hook *v1alpha1.ExternalAdmissionHook, i int) {
+			defer wg.Done()
+			if shouldCallHook(hook, attr) {
+				hookIndexCh <- i
+			}
+		}(&hooks[i], i)
+	}
+	wg.Wait()
+	close(hookIndexCh)
+
+	var hookIndice []int
+	for index := range hookIndexCh {
+		hookIndice = append(hookIndice, index)
+	}
+	if len(hookIndice) == 0 {
+		// no matching hooks
+		return nil
+	}
+
+	// convert the object to the external version before sending it to the webhook
+	var versioned versionedObjects
```

Comment:
What if you construct a new attr object instead of making this?

Reply:
Attr is an interface, so if we want to construct a new attr we would have to call `NewAttributesRecord`, and pass it all the `attr.GetObject()`, `attr.GetOldObject()`, etc., which is ugly.

Reply:
It's a lot easier than that thanks to embedding:

```go
type versioned struct {
  AttributesRecord
  oldObject runtime.Object
  object    runtime.Object
}

func (v versioned) GetObject() ...
func (v versioned) GetOldObject() ...
```

Not claiming you *should* do this, but it would mean 0 changes to the hook call.

Reply:
Good idea. How about revisiting this when I implement the mutating webhooks? Would want to make the hook call as similar as possible.

Reply:
Sure.

Reply:
Done.